apps/signing/views.py

```diff
@@ -0,0 +1,42 @@
+from django import forms
+
+from rest_framework import status
+from rest_framework.response import Response
+from tower import ugettext as _
+
+from api.jwt_auth.views import JWTProtectedView
+from addons.models import Addon
+from devhub.views import handle_upload
+from files.utils import parse_addon
+
+
+class UploadAddonView(JWTProtectedView):
+
+    def put(self, request, guid, version):
+        try:
+            addon = Addon.unfiltered.get(guid=guid)
+        except Addon.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        if not addon.has_author(request.user):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        filedata = request.FILES['upload']
+
+        # Verify that the guid and version match.
```

Comment:
It gets checked by `parse_addon`. It is really poorly named. It takes an optional `check` parameter that defaults to `True`. Maybe passing that in explicitly will make it more clear.